src/main/java/com/practice/slideshow/service/KafkaLoggingService.java

```diff
@@ -19,34 +21,40 @@
 @Slf4j
 public class KafkaLoggingService {
 
-  private final KafkaTemplate<String, String> kafkaTemplate;
+  private final KafkaTemplate<String, ImageResponse> kafkaTemplate;
 
   @Value("${app.kafka.topic}")
   private String topic;
 
   /**
-   * Logs a message to the Kafka topic.
+   * Logs an ImageResponse object to the Kafka topic using ProducerRecord.
    *
-   * @param actionType The type of action (e.g., "ADD_IMAGE", "DELETE_IMAGE").
-   * @param message    The log message.
+   * @param actionType Action type (e.g., "ADD_IMAGE", "DELETE_IMAGE", etc.)
+   * @param entity     The ImageEntity to be converted into ImageResponse.
    */
-  public void logAction(String actionType, String message) {
-    if (actionType == null || message == null) {
-      log.warn("Action type or message is null. Skipping log action.");
+  public void logAction(String actionType, ImageEntity entity) {
```

Comment:
it is not correct approach please use this link as an example https://github.com/dilipsundarraj1/kafka-for-developers-using-spring-boot/blob/main/library-events-producer/src/main/java/com/learnkafka/producer/LibraryEventProducer.java